events-api/src/modules/events/events.controller.ts

```diff
@@ -47,4 +58,54 @@ export class EventsController {
       data: foundEvent,
     };
   }
+
+  @Patch(':eventId')
+  @ApiOperation({ summary: 'Update event by ID' })
+  @ApiResponse({ status: 200, description: 'Event updated successfully.' })
+  @ApiResponse({ status: 404, description: 'Event not found.' })
+  async patchEventById(
+    @Param(
+      'eventId',
+      new ParseUUIDPipe({
+        errorHttpStatusCode: HttpStatus.BAD_REQUEST,
+        version: '4',
+      }),
+    )
+    eventId: string,
+    @Body() event: CreateEventReqDto,
+  ) {
+    const updatedEvent = await this.eventsService.updateEvent(eventId, event);
+    return updatedEvent;
+  }
+
+  @Delete(':eventId')
+  @ApiOperation({ summary: 'Delete event by ID' })
+  @ApiResponse({ status: 200, description: 'Event deleted successfully.' })
+  @ApiResponse({ status: 404, description: 'Event not found.' })
+  async deleteEventById(
+    @Param(
+      'eventId',
+      new ParseUUIDPipe({
+        errorHttpStatusCode: HttpStatus.BAD_REQUEST,
+        version: '4',
+      }),
+    )
+    eventId: string,
+  ) {
+    const deletedEvent = await this.eventsService.deleteEvent(eventId);
+    return {
+      deletedEvent
+    };
+  }
+
+
+@Get('/sort')
+@ApiOperation({ summary: 'Sort events' })
+@ApiResponse({ status: 200, description: 'Events sorted successfully.' })
+async sortEvents(@Query() query: SortEventsDto) {
```

Comment:
Why was a separate endpoint created? It's better to put it into the list events endpoint